### ISSUE TITLE
Cursor dissapears on server and client

### ISSUE THREAD

**Issue by jazomand** [debauchee/barrier#678]
### Operating Systems ###

Server: Windows 10 - 1909 (18363.836)

Client: MacOs 10.15.4

### Barrier Version ###

2.3.2-Release-210c2b70

### Steps to reproduce bug ###

1. Sleep MacOs computer while running Barrier.
2. Shutdown windows computer.
3. Turn on and login into windows computer (everything is fine in this moment, the mouse and keyboard are still working).
4. Wake MacOs computer. Then the cursos dissapears from both windows 10 and MacOs and you are not longer in control of any.

### Other info ###

* Is there a way to work around it? Yes. 
1. Physically disconnect the mouse from windows 10 and connect into MacOS. 
2. With that mouse on MacOS, stop and start Barrier (client). 
3. After this "restart", you can connect you mouse back to windows 10 (server).

* Does this bug prevent you from using Barrier entirely? Not yet, but is very annoying.

On the MacOS logs I only get "WARNING: cursor may not be visible".

**Comment by nikatjef** [comment]
Greetings,

  I run into this occasionally in Synergy and the basic problem is that the server is attempting to reset the mouse position and will occasionally reset it to the bottom left pixel of the client, but the client says the mouse is on the server.  When this happens, kill and restart the Synergy (Barrier) client.

**Comment by jazomand** [comment]
Yeah... that's what im doing, but is a valid issue to report, right?

**Comment by p12tic** [comment]
Let's not close valid bug reports.

**Comment by kar0v** [comment]
Also experiencing this on a new install between Windows server & Linux Mint client.

**Comment by tom-tan** [comment]
I have the same problem with Windows server (reproduced with Windows 10 & Windows 11) and macOS (11.6) client with the following barrier versions.
- Server (Windows): 2.3.3-release-dd8c2a1a (HEAD)
- Client (macOS): 2.3.3-release-dd8c2a1a (HEAD)

On the other hand, I found that it works with the following versions:
- Server (Windows): 2.3.3-release-3395cca9 ([release](https://github.com/debauchee/barrier/releases/tag/v2.3.3))
- Client (macOS):  2.3.3-release-dd8c2a1a (HEAD), 2.3.3-release-3395cca9 (release)

I guess it works with 2.3.3 release but does not work with HEAD.

**Comment by g1uc0se** [comment]
thanks to tom-tan,
I reproduce this invisble cursor by using 2.4.0 release on a windows(21H1) server, a windows(21H1) client and a mac client. After I change server version back to 2.3.3, barrier works perfect. I believe this is like you cannot use the same version on server and client bugs XD

**Comment by szeamer** [comment]
I am having exactly this problem on a Windows 10 server with a Windows 10 client. The client is a Microsoft Surface Book.
I have tried plugging my mouse into my laptop and back into the server machine - this did not solve the problem.
I have barrier version 2.4.0.

**Comment by szeamer** [comment]
A further note - whenever I try to move my cursor to the client screen, it seems to get stuck in the bottom right corner of the client screen.

**Comment by szeamer** [comment]
I have solved the issue for myself - my server was at 125% scale. I set it to 100% and now everything works as expected. 
More details on issue #1382. I might try downgrading to version 2.3.3 on my server to see if that allows me to put my server back at 125% scale. Will update here if that works.

**Comment by De-Coder-01** [comment]
> I have solved the issue for myself - my server was at 125% scale. I set it to 100% and now everything works as expected. More details on issue #1382. I might try downgrading to version 2.3.3 on my server to see if that allows me to put my server back at 125% scale. Will update here if that works.

Can you please update if downgrading solves the issue when scale is set to 125% on server. Thanks

**Comment by twadleigh** [comment]
I can confirm that setting the scale to 100% on my windows server fixed this issue for me with barrier 2.4.0.

I think this may be a regression because I just upgraded from 2.3.2, for which I didn't have a problem.

[I upgraded to 2.4.0 as part of upgrading my Ubuntu machine to 22.04, upgrading the version installed on my windows machine in parallel.]

**Comment by EsaKarjalainen** [comment]
Confirmed, changing server scaling worked. However, now I have eye strain.

**Comment by twadleigh** [comment]
> Confirmed, changing server scaling worked. However, now I have eye strain.

Yes, my _real_ workaround was to make my linux workstation the server. This was itself less than ideal for me because I actually preferred to use the thinkpad keyboard on my windows laptop. Still, I preferred to use a worse keyboard than make my display unusable.

**Comment by josephtesfaye** [comment]
Downloading the latest build from https://dev.azure.com/debauchee/Barrier/_build?definitionId=1 helped me.
